dolphin/db/sqlalchemy/api.py

```diff
@@ -298,17 +298,42 @@ def storage_get_all(context, marker=None, limit=None, sort_keys=None,
     return query.all()
 
 
+def storage_delete(context, storage_id):
+    """Delete a storage device."""
+    _storage_get_query(context).filter_by(id=storage_id).delete()
+
+
 def volume_get(context, volume_id):
     """Get a volume or raise an exception if it does not exist."""
     return NotImplemented
 
 
+def _volume_get(context, storage_id, session=None):
+    result = (_volume_get_query(context, session=session)
+              .filter_by(storage_id=storage_id)
+              .first())
+
+    if not result:
+        raise exception.NotFound()
```

Comment:
may be specific exception ?? like VolumeNotFound()

Reply:
This part of code is not used, so I removed them from this pr